### PR TITLE
Fix proguard settings

### LINKED ISSUE #5052: Crash when scanning a QR code on release build
#### Software and hardware versions 
Collect v2022.2-beta.0

#### Problem description
Scanning a QR code using a debug build works but with a release build a runtime exception is thrown: `class "com.github.fge.jsonschema.keyword.validator.common.AdditionalItemsValidator" has no appropriate constructor (keyword "additionalItems"; expected a constructor with a JsonNode as an argument)`

#### Steps to reproduce the problem
Scan any settings QR code with a release build.

#### Expected behavior
Scanning QR codes works with a release build.

#### Other information 
I addressed this for the beta by adding the following to the `collect_app` proguard rules:
`-keep class com.github.fge.jsonschema.keyword.validator.** { *; }`

It feels like this should be handled within the settings module, though.

Note: keeping `AdditionalItemsValidator` is not enough -- there's at least one other class that gets pruned.

### PR DESCRIPTION
**Pull request by seadowg**
Closes #5052, #5050

#### What has been done to verify that this works as intended?

Ran app and tests.

#### Why is this the best possible solution? Were any other approaches considered?

Comments inline.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

Main thing to check is importing settings (or creating a project) from a QR code as the underlying framework we use for validation has changed.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: buildSrc/src/main/java/dependencies/Dependencies.kt [hunk 1]
```diff
@@ -73,7 +73,7 @@ object Dependencies {
     const val slf4j_api = "org.slf4j:slf4j-api:2.0.0-alpha0"
     const val slf4j_timber = "com.arcao:slf4j-timber:3.1@aar"
     const val emoji_java = "com.vdurmont:emoji-java:5.1.1"
-    const val json_schema_validator = "com.github.java-json-tools:json-schema-validator:2.2.14"
+    const val json_schema_validator = "com.networknt:json-schema-validator:1.0.66"
```
Comment (seadowg) [inline buildSrc/src/main/java/dependencies/Dependencies.kt:76]: This framework is less "starred" on Github but actually seems to be more active and brings in a far less heavy set of dependencies (its README mentions this being intentional as well).

File: settings/consumer-rules.pro [hunk 2]
```diff
@@ -0,0 +1 @@
+-keep class com.github.fge.jsonschema.keyword.validator.** { *; }
```
Comment (seadowg) [inline settings/consumer-rules.pro:1]: This was the rule @lognaturel had needed in the `collect_app` proguard rules to release the beta.

File: settings/src/main/resources/settings_schema.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$schema": "https://json-schema.org/draft/2019-09/schema",
```
Comment (seadowg) [inline settings/src/main/resources/settings_schema.json:2]: https://github.com/networknt/json-schema-validator/ does not currently support the newest JSON schema version (`2020-12`). I think we should fix this as a contribution (issue here: https://github.com/networknt/json-schema-validator/issues/427) but right now it's not essential that we have it as far as I can see.

**Review by lognaturel — APPROVED** [review]

**Comment by kkrawczyk123** [comment]
**Tested with success!**
Verified on Androids: 5.1 and 8.1

Verified cases:
- compared size of apk PR 21,8 and 25,9 on the maser
- create project scanning qr code
- different types of servers: Central, Aggregate, Drive, Kobo, Ona 
- different settings combination
- update settings within the project through qr code
- import settings from file
- scan inverted qr code
- scan invalid qr code 
- use front barcode scanner

**Comment by srujner** [comment]
**Tested with success!**
Verified on Androids: 10 and 11

**Comment by lognaturel** [comment]
Is it possible this wasn't tried on a release build? I can't get it to work, I always get the "QR code does not contain valid settings" message. It works fine on a debug build. I'll take a look and file an issue if I can't figure it out.

File: settings/consumer-rules.pro (hunk 2 above)
Comment (lognaturel) [reply settings/consumer-rules.pro:1]: Whoops, this should now be for the new library, I think.

**Comment by seadowg** [comment]
@lognaturel as far as I remember I was **only** trying it on a release build (`selfSignedRelease` specifically). It's all manual though, so it's possible I messed something up.

**Comment by lognaturel** [comment]
Or maybe you first verified the proguard rule and then swapped out the library? Either way, I merged https://github.com/getodk/collect/pull/5066 myself and all is good.

@kkrawczyk123, @srujner I'm wondering whether you should just always use `selfSignedRelease`? That would have helped us catch some of these errors earlier. We can talk about it on Tues.

**Pull request merged**